src/microsoft/MicrosoftGraphSharePoint.js

```diff
@@ -0,0 +1,95 @@
+import { Client } from "@microsoft/microsoft-graph-client";
+import { OBOFAuthenticationProvider } from "./OBOFAuthenticationProvider.js";
+import { Util } from "../model/Util.js";
+
+export class MicrosoftGraphSharePoint {
+
+    static APPLICATION_JSON = 'application/json';
+
+    #siteId;
+    #defaultPath;
+    #client;
+    #url;
+    #folderId;
+
+    #tokenRefreshTried = false;
+    #obofAuthenticationProvider;
+
+    constructor(env) {
+        this.#siteId = env.SITE_ID;
+        this.#defaultPath = env.SITE_FILE_UPLOAD_PATH;
+        this.#folderId = env.SITE_FILE_UPLOAD_ID;
+        this.#url = `/sites/${this.#siteId}/drive`;
+        this.#obofAuthenticationProvider =  new OBOFAuthenticationProvider(env); 
+
+        this.#client = Client.initWithMiddleware({
+            debugLogging: true,
+            authProvider: this.#obofAuthenticationProvider
+        }); 
+    }
+
+    #getDefaultItem = async () => {
+        let response = await this.#client.api(`${this.#url}/root:/${this.#defaultPath}`).get();
+        return response;
+    }
+
+    #getItem = async (path) => {
+        let response = await this.#client.api(`${this.#url}/root:/${this.#defaultPath}/${path}`).get();
+        return response;
+    }
+
+    createFolder = async(folderName) => {
```

Comment:
why not use method syntax? (applies for all methods)

```suggestion
    sync createFolder(folderName) {
```